common/task/task.go

```diff
@@ -21,18 +21,32 @@
 package task
 
 import (
-	"github.com/uber/cadence/common"
+	"time"
 )
 
 type (
-	// SequentialTaskProcessor is the generic coroutine pool interface
-	// which process sequential task
-	SequentialTaskProcessor interface {
-		common.Daemon
-		Submit(task SequentialTask) error
+	// Info is the interface for describing a task
+	Info interface {
+		GetVersion() int64
+		GetTaskID() int64
+		GetTaskType() int
+		GetVisibilityTimestamp() time.Time
+		GetWorkflowID() string
+		GetRunID() string
+		GetDomainID() string
+	}
+
+	// SequenceIDs is a list of task SequenceID
+	SequenceIDs []SequenceID
+
+	// SequenceID determines the total order of tasks
+	SequenceID struct {
+		VisibilityTimestamp time.Time
+		TaskID              int64
 	}
```

Comment:
this is too specific, can we make a task key interface? basically ack manager does not need to understand this key interface implementation details, but have the ability to sort it for acking

Reply:
How about reusing the `task.Info` type as the key? The reason is that when implementing multi-cursor, each ackMgr needs additional information (beyond taskID and timestamp) about a task so that it knows how to split the outstanding tasks. Reusing also avoid unnecessarily type conversions.